README.md

```diff
@@ -578,6 +578,63 @@ This function is designed to rename a Delta table. It can operate either within
 rename_delta_table(existing_delta_table, "new_table_name")
 ```
 
+## Type 3 SCD Upserts
+
+Perform a type 3 scd upsert on a target delta table. 
+
+Parameters:
+
+- `delta_table` (`DeltaTable`): An object representing the delta table to be upserted.
+- `updates_df`  (`DataFrame`):  The data to be used in order to upsert the target delta table.
+- `primary_key`  (`str`):  The primary key (i.e. business key) uniquely identifiy each row in the target delta table.
+- `curr_prev_col_names`  (`dict[str,str]`): A dictionary of column names to store current and previous values.
```

Comment:
Can we separate this to two parameters: `current_col_name` and `previous_col_name`.  I generally prefer exposing public interfaces with full works instead of abbreviations.  Is there any advantage to grouping these two arguments in a dictionary?

Reply:
i think that, it depends on if we want to: 

1. Populate multiple columns at a time when applying the type 3 scd upserts:

Suppose that we have a delta table having this structure: 
+----+----+----+-------+--------+------------+-------------+--------------+
|pkey|name|job|prev_job| country|prev_country|    continent|prev_continent|
+----+----+---+--------+--------+------------+-------------+--------------+

--> The columns `prev_job`, `prev_country` and  `prev_continent` will store the previous values for the columns `job`, `country` and `continent` after applying the type 3 scd.

uisng a dictionary as a parameter to store the column names (for current and previous values) is less prone to error than storing them in two separate lists.

Option #1 
col_names = {"country":"prev_country", "job":"prev_job", "continent":"prev_continent"}
mack.type_3_scd_upsert(delta_table, updates_df, "pkey",col_names)

Option #2 
current_col_names = ["country","job","continent"]
previous_col_names = ["prev_country","prev_job","prev_continent"]
mack.type_3_scd_upsert(delta_table, updates_df, "pkey",current_col_names,previous_col_names)

Here an error on column order can cause a cartastrophy like storing the previous values of the column 'job' in the column 'prev_country' (or the opposite).
current_col_names = ["country","job","continent"]
previous_col_names = ["prev_job","prev_country","prev_continent"]


2. Populate one column at a time when applying the type 3 scd upserts 

let's use the same delta table structure mentioned above. 

We will have something like this: 

# type 3 scd upserts for column  `job`
current_col_names = "job"
previous_col_names = "prev_job"
mack.type_3_scd_upsert(delta_table, updates_df, "pkey",current_col_names,previous_col_names)

# type 3 scd upserts for column  `country`
current_col_names = "country"
previous_col_names = "prev_country"
mack.type_3_scd_upsert(delta_table, updates_df, "pkey",current_col_names,previous_col_names)

# type 3 scd upserts for column  `continent`
current_col_names = "continent"
previous_col_names = "prev_continent"
mack.type_3_scd_upsert(delta_table, updates_df, "pkey",current_col_names,previous_col_names)